packages/@aws-cdk/aws-lambda/lib/function-base.ts

```diff
@@ -209,14 +209,17 @@ export abstract class FunctionBase extends Resource implements IFunction {
     const action = permission.action || 'lambda:InvokeFunction';
     const scope = permission.scope || this;
 
-    new CfnPermission(scope, id, {
-      action,
-      principal,
-      functionName: this.functionArn,
-      eventSourceToken: permission.eventSourceToken,
-      sourceAccount: permission.sourceAccount,
-      sourceArn: permission.sourceArn,
-    });
+    // Skip duplicate permissions nodes. A more robust check would verify if the existing node matches the new one exactly.
+    if (!scope.node.tryFindChild(id)) {
```

Comment:
This doesn't feel like the right place to fix this bug. If two unaware 3rd party construct libraries call `fn.addPermission()` with the same `id`, the second will be silently ignored? Doesn't make sense.

Feels like the right place to fix this is at the higher level API where there is more context

Reply:
One other alternative I toyed with was fixing this just one level up, in the `grantInvoke` call. That feels like a slightly better compromise to me than making ELB have to deal with this. It makes reasonable sense to me that `grantInvoke` should be an idempotent operation, and not explode on duplicate calls. 

If the ELB code deals with this, the ELB code will have to have implementation-level details of how the Lambda grant is implemented, either by looping through all possible associated permissions or looking for the `Invoke${grantee}`-id'ed node, both of which seem like abstraction leakage to me.

Reply:
Maybe we can offer something like `fn.hasPermission(id)` or something to that affect to prevent the leakage?

Agreed that `grantInvoke()` should not explode on duplicate calls but we can't use the ID as the arbiter. It has to be the entire configuration.